spectec/src/il2al/translate.ml

```diff
@@ -10,12 +10,14 @@ open Source
 module Il = struct include Il include Ast include Print include Atom end
 
 
-(* Helpers *)
+(* Errors *)
+
+let error at msg = Error.error at "translation" msg
 
-let fail_on_exp exp typ_msg =
-  sprintf "Invalid expression `%s` to be %s."
-    (Il.Print.string_of_exp exp) typ_msg
-  |> failwith
+let error_exp exp typ =
+  error exp.at (sprintf "invalid %s: `%s`" typ (Il.Print.string_of_exp exp))
```

Comment:
In what sense is an expression "invalid"? Wouldn't "unexpected" or "unhandled" be more accurate?